lib/cron/synth-switch.ts

```diff
@@ -41,14 +79,178 @@ const handler: ScheduledHandler = async (_event: EventBridgeEvent<string, void>)
 
   let stmtId: string | undefined;
 
-  const configs = await readTokenConfig(log);
-  // TODO: format sql query string with configs
+  const configs = validateConfigs(await readTokenConfig(log));
+
+  // We can't pass in arrays as parameters to the query, so we have to build it into a formatted string
+  // inputToken and outputToken MUST be sanitized and lowercased before being passed into the query
+  const tokenInList = "('" + configs.map((config) => config.inputToken).join("', '") + "')";
+  const tokenOutList = "('" + configs.map((config) => config.outputToken).join("', '") + "')";
+
+  log.info(
+    {
+      tokenInList,
+      tokenOutList,
+      valueTokenInList: String(tokenInList),
+      valueTokenOutList: String(tokenOutList),
+    },
+    'formatted tokenInList, tokenOutList'
+  );
+
+  function hasPositiveTradeOutcome(order: ResultRowType): {
+    key: string;
+    result: boolean;
+  } {
+    const tradeType =
+      order.classic_amountin == order.classic_amountingasadjusted ? TradeType.EXACT_INPUT : TradeType.EXACT_OUTPUT;
+    const trade: SynthSwitchRequestBody = {
+      inputToken: order.tokenin,
+      inputTokenChainId: parseInt(order.tokeninchainid),
+      outputToken: order.tokenout,
+      outputTokenChainId: parseInt(order.tokenoutchainid),
+      type: String(tradeType),
+      amount: tradeType == TradeType.EXACT_INPUT ? order.classic_amountin : order.classic_amountout,
+    };
+    const key = SwitchRepository.getKey(trade);
+    let hasPriceImprovement: boolean;
+    if (tradeType == TradeType.EXACT_INPUT) {
+      hasPriceImprovement = BigNumber.from(order.settledAmountOut).gt(order.classic_amountoutgasadjusted);
+    } else {
+      hasPriceImprovement = BigNumber.from(order.classic_amountingasadjusted).gt(order.settledAmountIn);
+    }
+    // can add more conditionals here
+    const result = hasPriceImprovement;
+    return { key, result };
+  }
+
+  async function updateSynthSwitchRepository(configs: TokenConfig[], result: ResultRowType[]) {
```

Comment:
if we do all the calculations in lambda sequentially, which I think is fine for a cron job, we should give it enough memory/timeout threshold - not sure if current setting needs a change tho justing calling out

Reply:
smart yeah, increasing timeout to 10 min and will reduce after testing